agents/alm.py

```diff
@@ -311,25 +298,31 @@ def _intrinsic_reward_loss(self, z, action_batch, z_next, z_next_prior, log, met
     def update_critic(self, z_batch, action_batch, reward_batch, z_next_batch, discount_batch, std, log, metrics):
         with torch.no_grad():
             next_action_dist = self.actor(z_next_batch, std)
-            next_action_batch = next_action_dist.rsample(clip=self.stddev_clip)
-
+            next_action_batch = next_action_dist.sample(clip=self.stddev_clip)
+
+        if self.critic_mode == 'model':
+            Q1, Q2, R = self.critic(z_batch, action_batch)
+            Q1_, Q2_, R_ = self.critic(z_next_batch, next_action_batch)
```

Comment:
Do not change the code that uses critic, change only the critic itself. This way you will make less errors.
```
Q1 = Q1 * self.gamma + R
``` 
should be inside the ModelCritic class.

E.g. in this case you use `critic` where the original code used  critic_target, which is an error that you wouldn't make following this advice.